transport protocol specification (#471)
vacp2p/rfc: content/docs/rfcs/10/README.md

> 

Reviewer: Add your name to the contributor list at the top :)

Author: resolved in 3c212ea20df7ffb3ee009d1dce5b11fec4537a2c

See [23/WAKU2-TOPICS](/spec/23) for more information about recommended topic usage.

> # Transport Protocol Specification

Reviewer: ```suggestion
# Transports
```

Author: done in ea014867f9062d531dd9f1772c99e4b2058f5f80

> This section describes the specification for transport protocols supported by waku v2. While Waku strives to be transport agonistic, however any underlying transport layer MUST be a reliable transport.

Reviewer 2: ```suggestion
This section describes the specification for transport protocols supported by Waku v2. While Waku v2 strives to be transport agonistic, however any underlying transport layer MUST be a reliable transport.
```

Reviewer: ```suggestion
Waku v2 is built in top of libp2p, and like libp2p it strives to be transport agnostic.
We define a set of recommended transports in order to achieve a baseline of interoperability between clients.
This section describes these recommended transports.
```

Reviewer: - Soften requirement on reliable transport as that's not necessarily set in stone and limits us needlessly
- Reword and be more clear on rationale

Author: done ea014867f9062d531dd9f1772c99e4b2058f5f80

> Waku client implementations SHOULD support the TCP transport, and MUST be enabled for both dialing and listening even if other transports are available if TCP is available.

Reviewer: We try to use https://sembr.org/

Reviewer: I'd add a sentence here saying that implementations running in certain restricted environments, such as browsers, may not be able to use TCP directly.

Reviewer 3: ```suggestion
Waku client implementations SHOULD support the TCP transport.
Where TCP is supported it MUST be enabled for both dialing and listening, even if other transports are available.
```

Reviewer 3: And agree with adding a comment on environments where TCP might not be appropriate.

Author: done in 3c212ea20df7ffb3ee009d1dce5b11fec4537a2c

> The underlying transports SHOULD support secure websocket and MAY support websockets for bidirectional communication streams and web browser context.

Reviewer 2: ```suggestion
The underlying transports SHOULD support secure websocket and MAY support insecure websockets for bidirectional communication streams and web browser context.
```
following one of the @jm-clius's comment

Reviewer 3: Maybe split into two sentences to make it clearer under what conditions `unsecure websockets` could be used.
Something like:
```suggestion
A Waku v2 node SHOULD support secure websockets for bidirectional communication streams, for example in a web browser context.
A node MAY support unsecure websockets if required by the application or running environment.
```

Author: done in ea014867f9062d531dd9f1772c99e4b2058f5f80

See [23/WAKU2-TOPICS](/spec/23) for more information about recommended topic usage.

# Transports

Waku v2 is built in top of libp2p, and like libp2p it strives to be transport agnostic.
We define a set of recommended transports in order to achieve a baseline of interoperability between clients.

This section describes these recommended transports.

Waku client implementations SHOULD support the TCP transport.

Where TCP is supported it MUST be enabled for both dialing and listening, even if other transports are available.

A Waku v2 node SHOULD support secure websockets for bidirectional communication streams, for example in a web browser context.

> A node MAY support unsecure websockets if required by the application or running environment.

Reviewer: ```suggestion
A node MAY support unsecure websockets if required by the application or running environment.

```
